app/src/main/cpp/skyline/applet/error_applet.cpp

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2022 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#include "error_applet.h"
+#include "services/am/storage/ObjIStorage.h"
+
+namespace skyline::applet {
+    ErrorApplet::ErrorApplet(const DeviceState &state,
+                             service::ServiceManager &manager,
+                             std::shared_ptr<kernel::type::KEvent> onAppletStateChanged,
+                             std::shared_ptr<kernel::type::KEvent> onNormalDataPushFromApplet,
+                             std::shared_ptr<kernel::type::KEvent> onInteractiveDataPushFromApplet,
+                             service::applet::LibraryAppletMode appletMode)
+        : IApplet{state, manager, std::move(onAppletStateChanged), std::move(onNormalDataPushFromApplet), std::move(onInteractiveDataPushFromApplet), appletMode} {}
+
+    Result ErrorApplet::Start() {
+        auto commonArg{PopNormalInput<service::applet::CommonArguments>()};
+
+        errorStorage = PopNormalInput();
+        auto errorCommonHeader{errorStorage->GetSpan().as<ErrorCommonHeader>()};
+        Logger::Debug("ErrorApplet: version: 0x{:X}, type: 0x{:X}", commonArg.apiVersion, errorCommonHeader.type);
+
+        switch (errorCommonHeader.type) {
+            case ErrorType::ApplicationErrorArg:
+                HandleApplicationErrorArg();
+                break;
+            default:
+                throw exception("ErrorApplet type 0x{:X} is not implemented", errorCommonHeader.type);
+        }
+
+        // Notify the guest that we've finished running
+        onAppletStateChanged->Signal();
+
+        return {};
+    }
+
+    void ErrorApplet::HandleApplicationErrorArg() {
+        auto applicationErrorStorage{errorStorage->GetSpan().as<ApplicationErrorArg>()};
+
+        Logger::Error("{}\nFull message: {}", applicationErrorStorage.dialogMessage.data(), applicationErrorStorage.fullscreenMessage.data());
```

Comment:
Isn't it possible for `Full Message` to be unset? It shouldn't really be printed in that case. I'd also recommend `ErrorNoPrefix("Application Error: ...")` as it'd be better than `HandleApplicationErrorArg`.

Reply:
As a stub, I'd say it's not too bad to print `Full message:` even when one isn't supplied. Conditionally appending that would be annoying IMO as you'd have to format locally, then submit to logger. We can do this properly later on.

Reply:
That's not required, it can simply be:
```suggestion
    if (applicationErrorStorage.fullscreenMessage.empty())
        Logger::ErrorNoPrefix("Application Error: {}", applicationErrorStorage.dialogMessage.data());
    else
        Logger::ErrorNoPrefix("Application Error: {}\nFull message: {}", applicationErrorStorage.dialogMessage.data(), applicationErrorStorage.fullscreenMessage.data());
```

Reply:
That's not required, it can simply be:
```suggestion
    if (applicationErrorStorage.fullscreenMessage.empty())
        Logger::ErrorNoPrefix("Application Error: {}", applicationErrorStorage.dialogMessage.data());
    else
        Logger::ErrorNoPrefix("Application Error: {}\nFull message: {}", applicationErrorStorage.dialogMessage.data(), applicationErrorStorage.fullscreenMessage.data());
```

Reply:
Well I didn't think of that for some reason, lgtm!